great_expectations/datasource/fluent/interfaces.py

```diff
@@ -304,11 +284,13 @@ def add_batch_config(
         batch_config = BatchConfig(name=name, partitioner=partitioner)
         batch_config.set_data_asset(self)
         self.batch_configs.append(batch_config)
+        self.update_batch_config_field_set()
```

Comment:
We now need to always remember to call this after each update to `self.batch_configs`? Is there a better way to do this without putting the burden on future devs?

Reply:
cc @tyler-hoffman  - this was a bugfix he snuck in, since the context was in this PR

Reply:
i've added [a ticket](https://greatexpectations.atlassian.net/browse/V1-206), and will have a solution as a followup 🙇